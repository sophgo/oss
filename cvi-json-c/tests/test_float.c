/* Copyright (C) 2016 by Rainer Gerhards
 * Released under ASL 2.0 */
#include "config.h"
#include "cvi_json_object.h"
#include "cvi_json_tokener.h"
#include <stdio.h>
int main(void)
{
	cvi_json_object *cvi_json;

	cvi_json = cvi_json_object_new_double(1.0);
	printf("cvi_json = %s\n", cvi_json_object_to_cvi_json_string_ext(cvi_json, JSON_C_TO_STRING_PRETTY));
	cvi_json_object_put(cvi_json);

	cvi_json = cvi_json_object_new_double(-1.0);
	printf("cvi_json = %s\n", cvi_json_object_to_cvi_json_string_ext(cvi_json, JSON_C_TO_STRING_PRETTY));
	cvi_json_object_put(cvi_json);
	cvi_json = cvi_json_object_new_double(1.23);
	printf("cvi_json = %s\n", cvi_json_object_to_cvi_json_string_ext(cvi_json, JSON_C_TO_STRING_PRETTY));
	cvi_json_object_put(cvi_json);
	cvi_json = cvi_json_object_new_double(123456789.0);
	printf("cvi_json = %s\n", cvi_json_object_to_cvi_json_string_ext(cvi_json, JSON_C_TO_STRING_PRETTY));
	cvi_json_object_put(cvi_json);
	cvi_json = cvi_json_object_new_double(123456789.123);
	printf("cvi_json = %s\n", cvi_json_object_to_cvi_json_string_ext(cvi_json, JSON_C_TO_STRING_PRETTY));
	cvi_json_object_put(cvi_json);
	return 0;
}
