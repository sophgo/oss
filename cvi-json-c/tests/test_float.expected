cvi_json = 1.0
cvi_json = -1.0
cvi_json = 1.23
cvi_json = 123456789.0
cvi_json = 123456789.123
